tests/util/yaml/test_secrets.py

```diff
@@ -0,0 +1,163 @@
+    """Did secrets load ok."""
+    loaded_file = load_config_file(
+        default_config.path, [default_config, default_secrets]
+    )
+    expected = {"api_password": "pwhttp"}
+    assert expected == loaded_file["http"]
+
+    expected = {"username": "un1", "password": "pw1"}
+    assert expected == loaded_file["component"]
+
+
+def test_secrets_from_parent_folder(filepaths, default_config, default_secrets) -> None:
+    """Test loading secrets from parent folder."""
+    config_file = YamlFile(
+        path=filepaths["sub_folder"] / "sub.yaml",
+        contents=default_config.contents,
+    )
+    loaded_file = load_config_file(config_file.path, [config_file, default_secrets])
+    expected = {"api_password": "pwhttp"}
+
+    assert expected == loaded_file["http"]
+
+
+def test_secret_overrides_parent(filepaths, default_config, default_secrets) -> None:
+    """Test loading current directory secret overrides the parent."""
+    config_file = YamlFile(
+        path=filepaths["sub_folder"] / "sub.yaml", contents=default_config.contents
+    )
+    sub_secrets = YamlFile(
+        path=filepaths["sub_folder"] / yaml.SECRET_YAML, contents="http_pw: override"
+    )
+
+    loaded_file = load_config_file(
+        config_file.path, [config_file, default_secrets, sub_secrets]
+    )
+
+    expected = {"api_password": "override"}
+    assert loaded_file["http"] == expected
+
+
+def test_secrets_from_unrelated_fails(filepaths, default_secrets) -> None:
+    """Test loading secrets from unrelated folder fails."""
+    config_file = YamlFile(
+        path=filepaths["sub_folder"] / "sub.yaml",
+        contents="http:\n  api_password: !secret test",
+    )
+    unrelated_secrets = YamlFile(
+        path=filepaths["unrelated"] / yaml.SECRET_YAML, contents="test: failure"
+    )
+    with pytest.raises(HomeAssistantError, match="Secret test not defined"):
+        load_config_file(
+            config_file.path, [config_file, default_secrets, unrelated_secrets]
+        )
+
+
+def test_secrets_logger_removed(filepaths, default_secrets) -> None:
+    """Ensure logger: debug was removed."""
+    config_file = YamlFile(
+        path=filepaths["config"] / YAML_CONFIG_FILE,
+        contents="api_password: !secret logger",
+    )
+    with pytest.raises(HomeAssistantError, match="Secret logger not defined"):
+        load_config_file(config_file.path, [config_file, default_secrets])
+
+
+@patch("homeassistant.util.yaml.loader._LOGGER.error")
+def test_bad_logger_value(mock_error, filepaths) -> None:
+    """Ensure logger: debug was removed."""
+    config_file = YamlFile(
+        path=filepaths["config"] / YAML_CONFIG_FILE, contents="api_password: !secret pw"
+    )
+    secrets_file = YamlFile(
+        path=filepaths["config"] / yaml.SECRET_YAML, contents="logger: info\npw: abc"
+    )
+    load_config_file(config_file.path, [config_file, secrets_file])
+    assert mock_error.call_count == 1, "Expected an error about logger: value"
```

Comment:
I'd instead use the `caplog` fixture to assert logs.

https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-caplog